tools/engine_tool/lib/src/build_utils.dart

```diff
@@ -48,3 +48,15 @@ List<Build> runnableBuilds(Environment env, Map<String, BuilderConfig> input) {
     return build.canRunOn(env.platform);
   });
 }
+
+/// Validates the list of builds.
+/// Calls assert.
```

Comment:
I am curious when in the product lifecycle we'll run in asserts enabled mode?

I'd either make this a throw or just log a message personally

Reply:
Until we have a better UI for specifying configs we can't disambiguate duplicates. If this assertion ever fired it would indicate our checked-in data is bad. 

I'm fine with doing this in all build modes if people would rather not use assert()

Reply:
I don't feel strongly. I'm fine with using `assert`s but in this case `throw`ing seems reasonable too.